conans/test/functional/compression/compression_test.py

```diff
@@ -0,0 +1,458 @@
+import abc
+import ctypes
+import os
+import random
+import six
+import stat
+import string
+import sys
+import tarfile
+import time
+import uuid
+import zipfile
+
+import unittest
+from nose.plugins.attrib import attr
+from parameterized import parameterized_class
+
+from conans.client.tools.files import chdir, load
+from conans.util.files import md5sum
+from conans.test.utils.tools import TestClient
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Compressor:
+    @abc.abstractmethod
+    def compress(self, d):
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def decompress(self, d):
+        raise NotImplementedError()
+
+
+class ZipCompressor(Compressor):
+    ext = ".zip"
+
+    def compress(self, d):
+        def symlink_zipinfo(name):
+            zi = zipfile.ZipInfo(name)
+            # external_attr is 4 bytes in size
+            # top 4 bits are for file type, 4 bits for setuid/setguid, then 16 bits for permissions
+            # low order byte is MS-DOS directory attribute
+            # 0xA000 (or 0120000 octal) is traditional value for symbolic link file type (S_IFLNK)
+            # see also https://trac.edgewall.org/attachment/ticket/8919/ZipDownload.patch
+            permissions = 0xA000
+            zi.create_system = 3
+            zi.external_attr = permissions << 16
+            return zi
+
+        with zipfile.ZipFile(file='test' + self.ext, mode='w', compression=zipfile.ZIP_DEFLATED) as f:
+            for root, dirnames, filenames in os.walk(d):
+                for filename in filenames:
+                    filename = os.path.join(root, filename)
+                    archname = os.path.relpath(filename, d)
+                    if os.path.islink(filename):
+                        target = os.readlink(filename)
+                        f.writestr(symlink_zipinfo(archname), target)
+                    else:
+                        f.write(filename, archname)
+                for dirname in dirnames:
+                    filename = os.path.join(root, dirname)
+                    archname = os.path.relpath(filename, d)
+                    if os.path.islink(filename):
+                        target = os.readlink(filename)
+                        f.writestr(symlink_zipinfo(archname), target)
+                    else:
+                        f.write(filename, archname)
+            f.close()
```

Comment:
Is it required? Is it not closed after the context manager?

Reply:
removed